index.html

```diff
@@ -955,6 +987,9 @@ <h2>Verifiable Credential</h2>
               <li>Any empty value MUST be represented as a null value.</li>
               <li>Any other value MUST be represented as a String type.</li>
             </ul>
+           <p> Certain properties must be single valued, whilst other properties 
+ 	can be an array of values, or a set of objects. The examples and text make it clear which properties can have which types of value.
```

Comment:
Hmm, except that this is not enforced by the lower data model. Don't know what to do here, I understand the desire, but getting that working would require an additional layer of JSON Schema that adds complexity and that we probably don't want to impose on developers.

Reply:
What is this lower data model you are referring to? Surely the VC data model should be self contained, or, if it refers to other standards, make it clear what other standards it relies upon. Personally I think that when the IETF JSON Schema spec is completed, we should define a schema for verifiable credentials, otherwise you are much less likely to get interworking. X.509 had a tightly defined schema, with clear extension points, but even this was too broad to guarantee interworking, which gave rise the PKIX working group being created. We should not develop a VC data model that essentially says any JSON construct can be put anywhere in a VC, as that is bound to guarantee no interworking, and in the worst case the death of VCs, or in the best case the spawning of countless other standards or profiles. (Look where we are with X.509 and PKIX with other 50 more standards and dreadful interworking - I am the co-author of several papers on this if you want the details)

Reply:
> We should not develop a VC data model that essentially says any JSON construct can be put anywhere in a VC...

I agree. I think this is what @msporny is alluding to when referring to the "lower data model". We have a concept of the generic data model functioning as a graph (or as a tree). We mention this graph when talking about the information related to subjects but it sounds like we should be more explicit about how the entire data model can be viewed consistently that way, which is a powerful mechanism for informing people as to how to extend it in a compliant fashion. In fact, it now seems rather glaring to me that we don't have an extensibility section in the spec at all.

Reply:
@dlongley I am glad that you agree. But having the "type": construct as the topmost data type, without any restrictions on what types can be placed there, is essentially an unconstrained structure that allows any data type to be defined. Pulling out the two types (VC and VP) into JSON objects, constrains our data model to only support VCs and VPs (for now).
"In fact, it now seems rather glaring to me that we don't have an extensibility section in the spec at all." I would actually say that the converse is currently true, that we do not constrain what can be placed in most places in the data model.